File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        return self.getTree(nums,0,len(nums)-1)
    def getTree(self,nums,start,end):
        if start > end:
            return None
        mid = (start+end)//2
        root = TreeNode()
        root.val = nums[mid]
        root.left = self.getTree(nums,start,mid-1)
        root.right = self.getTree(nums,mid+1,end)
        return root
        

File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return getTree(nums,0,nums.length-1);
    }
    public TreeNode getTree(int[] nums,int start,int end) {
        if(start>end)
            return null;
        int mid = (start+end)/2;
        TreeNode root = new TreeNode();
        root.val = nums[mid];
        root.left = getTree(nums,start,mid-1);
        root.right = getTree(nums,mid+1,end);
        return root;
    }
}